moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -657,16 +661,23 @@ class LockedPlanningSceneRO
 
   operator bool() const
   {
-    return planning_scene_monitor_ && planning_scene_monitor_->getPlanningScene();
+    planning_scene_monitor::LockedPlanningSceneRO ls(planning_scene_monitor_);
+    return planning_scene_monitor_ && ls;
   }
 
   operator const planning_scene::PlanningSceneConstPtr&() const
   {
+// Ignore getPlanningScene() deprecation
```

Comment:
It makes absolutely no sense to deprecate a function but then instead of adapting the code to only suppress the deprecation warnings. This needs to be fixed. Obviously, for this kind of locking feature we still need access to the internal planning scene, we just don't want to expose it publicly. One solution: deprecate the old function, make the `LockedPlanningScene*` classes friend classes to allow them access to PSM members and private functions, (optional but recommended) provide a new private function `PlanningSceneMonitor::getInternalPlanningScene()` and use that instead.

I'm still not entirely sure about the possible side effects of this deprecation, and will do some more reviewing before approval. We should also discuss if the public `PlanningSceneMonitor::lockScene*()` functions should be deprecated as well with this change because there is no use for them anymore.

Reply:
`provide a new private function PlanningSceneMonitor::getInternalPlanningScene()`

I don't think we should do this because that might encourage future MoveIt developers to use it in the future. We don't want that.

Reply:
> We should also discuss if the public PlanningSceneMonitor::lockScene*() functions should be deprecated as well with this change because there is no use for them anymore.

My thinking is, we ought to keep those methods because they have other functionality, like this:

```
void PlanningSceneMonitor::lockSceneRead()
{
  scene_update_mutex_.lock_shared();
  if (octomap_monitor_)
    octomap_monitor_->getOcTreePtr()->lockRead();
}
```

Reply:
`One solution: ...`

Done!

Reply:
> `provide a new private function PlanningSceneMonitor::getInternalPlanningScene()`
> 
> I don't think we should do this because that might encourage future MoveIt developers to use it in the future. We don't want that.

They can't, that's what private is for. And if they do, they have do it explicitly with a friend class like here, in which case I trust they have a good reason, like here.